source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -56,6 +77,7 @@ void ns_dyn_mem_init(uint8_t *heap, uint16_t h_size, void (*passed_fptr)(heap_fa
         heap += (sizeof(int) - temp_int);
         h_size -= (sizeof(int) - temp_int);
     }
+    memset(&holes_list, 0, sizeof(holes_list_t));
```

Comment:
No no no. Zero-initting doesn't do anything useful, and you're initting it below anyway.